src/main/java/org/traccar/protocol/TeltonikaProtocolDecoder.java

```diff
@@ -538,7 +540,7 @@ private void decodeLocation(Position position, ByteBuf buf, int codec) {
                 int length = buf.readUnsignedShort();
                 if (id == 256) {
                     position.set(Position.KEY_VIN, buf.readSlice(length).toString(StandardCharsets.US_ASCII));
-                } else if (id == 385) {
+                } else if ((id == 385) || (id == 10500)) {
```

Comment:
You don't need nested brackets.

Reply:
Feel free to remove the brackets - it will not change much I think. Some like brackets some don't.

Reply:
Just to clarify, are you interested in fixing this PR or do you want us to take over?

Reply:
Take over please. It works for me like this - perhaps I have different java version. You are the one that could make it work for everybody.